generator/sbpg/targets/resources/sbp_messages_mod.rs

```diff
@@ -22,19 +22,19 @@ use self::(((p.identifier|mod_name)))::(((m.identifier|camel_case)));
 ((*- endfor *))
 use self::unknown::Unknown;
 
-#[cfg(feature = "sbp_serde")]
-use serde::{Serialize, Deserialize};
 use crate::serialize::SbpSerialize;
-use crate::framer::FramerError;
 
+#[enum_dispatch::enum_dispatch]
 pub trait SBPMessage: SbpSerialize {
     fn get_message_type(&self) -> u16;
     fn get_sender_id(&self) -> Option<u16>;
     fn set_sender_id(&mut self, new_id: u16);
-    fn to_frame(&self) -> std::result::Result<Vec<u8>, FramerError>;
+    fn to_frame(&self) -> std::result::Result<Vec<u8>, crate::FramerError>;
+    fn write_frame(&self, buf: &mut Vec<u8>) -> std::result::Result<(), crate::FramerError>;
 }
 
-#[cfg_attr(feature = "sbp_serde", derive(Serialize, Deserialize))]
+#[enum_dispatch::enum_dispatch(SbpSerialize, SBPMessage)]
+#[cfg_attr(feature = "sbp_serde", derive(serde::Serialize), serde(untagged))]
```

Comment:
Only derive `Serialize` because parsing of sbp json is done with manually based on the base64 payload